optimization/ht_functions.py

```diff
@@ -0,0 +1,385 @@
+            print("The range you have selected does not satisfy pressure drop\
+ requirements, consider increasing coolant flow channel diameter")
+            sys.exit()
+
+        not_AR = 1
+        if self.select_AR == True:
+            not_AR = np.nan
+
+        # calculate departure from AR = 1
+        aspect_err = np.subtract(self.data['AR'], np.ones([self.N, self.N]))
+        aspect_err = np.abs(aspect_err)
+        # select AR error no greater than 0.33 (based on NuScale AR = 1.33)
+        on_off = lambda x: not_AR if x > 0.33 else 1
+        vfunc = np.vectorize(on_off)
+
+        for key in self.data:
+            self.data[key] = np.multiply(vfunc(aspect_err), self.data[key])
+        
+        if False not in np.isnan(self.data['AR']):
+            print("The range you have selected does not satisfy aspect ratio\
+ requirements, consider a new geometry range")
+            sys.exit()
+        
+        # search the results for minimum-mass configuration
+        mindices_mass = np.unravel_index(
+                   np.nanargmin(self.data['mass']), self.data['mass'].shape)
+        # get data for optimal AR
+        self.min_idx, self.min_jdx = mindices_mass
+        self.min_mass = self.data['mass'][self.min_idx][self.min_jdx]
+
+        # save the optimal configuration
+        self.minD = self.D[self.min_idx][self.min_jdx]
+        self.minPD = self.PD[self.min_idx][self.min_jdx]
+        
+        # report the optimal configuration and it's corresponding fuel mass 
+        outstring =  "1D Thermal Hydraulics Optimization Results:\n"
+        outstring += "Reactor minimum mass (m = " + str(round(self.min_mass,3))\
+        + "[kg]) " + " occurs at D = " + str(self.minD) +  "[m] & PD = "\
+        + str(self.minPD) + "[-]."
+        print(outstring)
+
+    def plot(self, D, PD, key):
+        """Produce surface plot of the flow results as function of PD and coolant
+        channel diameter.
+        """
+        # get parametric sweep data
+        M = self.data[key]
+        fig = plt.figure()
+        ax = fig.gca(projection='3d')
+        surf = ax.plot_surface(D, PD, M, 
+                               cmap=cm.viridis, linewidth=0,
+                               vmin=0, vmax=np.nanmax(M),
+                               antialiased=False)
+
+        # set x/y axis labels, ticks
+        ax.set_xlabel("Coolant Channel Diameter [m]", fontsize=7)
+        plt.xticks(rotation=25, fontsize=6)
+        ax.set_ylabel("Fuel Pitch to Coolant D Ratio [-]", fontsize=7)
+        plt.yticks(rotation=25, fontsize=6)
+        ax.set_zlabel(self.titles[key][1], fontsize=7)
+         
+        # Customize the z axis.
+        ax.set_zlim(np.nanmin(M),np.nanmax(M))
+        ax.zaxis.set_major_locator(LinearLocator(10))
+        ax.zaxis.set_major_formatter(FormatStrFormatter('%.02f'))
+        
+        # edit z tick labels
+        for t in ax.zaxis.get_major_ticks(): t.label.set_fontsize(6)
```

Comment:
PEP8: "Compound statements (multiple statements on the same line) are generally discouraged."
-> break the line